src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -256,6 +257,24 @@ private function getNormalizer(mixed $data, ?string $format, array $context): ?N
                     continue;
                 }
 
+                if ($normalizer instanceof SupportedTypesMethodInterface && (null !== $supportedTypes = $normalizer->getSupportedTypes())) {
+                    foreach ($supportedTypes as $supportedType => $isCacheable) {
+                        if ($type !== $supportedType && !is_subclass_of($type, $supportedType, true)) {
+                            continue;
+                        }
+
+                        if ($isCacheable && $normalizer->supportsNormalization($data, $format, $context)) {
```

Comment:
If tgetSupportedTypes would already return the types for this specific format, like;

```php
getSupportedTypes(string $format = null, array $context = [])
```

Then this second check wouldn't even be needed, because it would always return true:

```suggestion
                        if ($isCacheable) {
```

Reply:
`getSupportedTypes` must *not* be context-sensitive, or we'll get back to non-cacheable results, voiding the perf improvement.

Reply:
Actually my suggestion doesn't need `$context` (the same way `->supportsNormalization` shouldn't need it in this line since its return value will be cached, afaik). I will edit my example snippet.